frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/OptionListEditor/OptionListEditor.tsx

```diff
@@ -48,137 +51,28 @@ export function OptionListEditor({
     case 'success': {
       if (optionsLists[optionsId] !== undefined) {
         return (
-          <EditLibraryOptionListEditorModal
-            label={label}
+          <LibraryOptionsEditor
+            component={component}
             optionsId={optionsId}
             optionsList={optionsLists[optionsId]}
+            handleDelete={handleDelete}
           />
         );
-      }
-      if (component.options !== undefined) {
+      } else if (component.options !== undefined) {
         return (
-          <EditManualOptionListEditorModal
-            label={label}
+          <ManualOptionsEditor
             component={component}
             handleComponentChange={handleComponentChange}
+            handleDelete={handleDelete}
           />
         );
+      } else {
+        return (
+          <StudioAlert severity={'info'} size='sm'>
+            {t('ux_editor.options.tab_code_list_alert_title')}
```

Comment:
Will this alert be shown if the optionList on the component does not exist in the app and if manual options are not set? From the text it looks like there should be a check that the optionListId that is set is a custom one, i.e. one that is not in the list of optionListIds.

Reply:
The alert should only be shown when there is a `component.optionsId` which does not correspond with any of the code lists from the library and `component.options` is undefined

Reply:
Yes. My point is that to make it clear when the alert should be shown we should have a dedicated function that checks the conditions; there is no manual options set and the connected optionsId is not in the list of the option lists belonging to the app.

And alo, as I commented below, we should consider using the same pattern as for images here. Meaning that you are able to swap between the tabs, but you will get an alert if you open a tab when there are configs on the component belonging to another tab. And if you change something in the current tab, you will override the old config.

But I think it would make more sense implementing this after the singel-option-list-fetching is merged 🤔